### PR TITLE
Update OTEL client to 1.33 and Go to 1.23

### PR DESCRIPTION
**Pull request by ryanrolds**
It also includes some fixes to address test failures.

### CODE REVIEW

**Comment by ryanrolds** [comment]
I'm looking into some things.

**Comment by mdelapenya** [comment]
@ryanrolds may I add some commits bumping the version of testcontainers-go, so that the tests are more accurate?

**Comment by ryanrolds** [comment]
> @ryanrolds may I add some commits bumping the version of testcontainers-go, so that the tests are more accurate?

Feel free. I think this is a in an OK state. Please review. :)

**Review by mdelapenya — APPROVED** [review]
@ryanrolds LGTM, thanks! Just one comment regarding a debug log.

This PR motivated me to go back to the project, so sending my personal thanks for it.

File: main_test.go [hunk 1]
```diff
@@ -214,6 +214,16 @@ func Test_Main_SampleXML(t *testing.T) {
...
+		ports, portErr := otelCollector.Ports(ctx)
+		if portErr != nil {
+			fmt.Printf("could not get ports for otel-collector: %v", portErr)
+			t.Errorf("could not get ports for otel-collector: %v", portErr)
+		}
+
+		for _, port := range ports {
+			fmt.Printf("port: %s\n", port)
+		}
+
```
[the remark this answers is not in the document]
Comment (ryanrolds) [reply main_test.go:226]: Yes, sorry. That was me debugging random test failures.

**Comment by ryanrolds** [comment]
No problem. :) More PRs incoming. I've got this working in our CI and I'm working on reducing cardinality and adding more metrics. I've noticed an issue with "duration"/time being zero. I may be making a change to junit-go to correct the issue, I'm still investigating.

File: main_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ryanrolds) [reply main_test.go]: Removed.

**Comment by ryanrolds** [comment]
Are you OK with me merging? How do you want to handle that? My only concern is the test hasn't run on the latest change.

File: main_test.go [hunk 2]
```diff
@@ -212,9 +213,17 @@ func Test_Main_SampleXML(t *testing.T) {
 		t.Error(err)
 	}
 
-	collectorPort, err := otelCollector.MappedPort(ctx, "4317/tcp")
+	var collectorPort nat.Port
+	err = retry.Do(func() error {
```
Comment (mdelapenya) [inline main_test.go:217]: Suggestion: Let's not fix this here with the retry, and wait for my bump of the Testcontainers library. Can you remove the import too?

File: main_test.go [hunk 3]
```diff
@@ -251,25 +260,31 @@ func Test_Main_SampleXML(t *testing.T) {
 		t.Error()
 	}
 
-	// TODO: retry until the file is written by the otel-exporter
```
Comment (mdelapenya) [inline main_test.go:254]: Suggestion: I have this fixed in the TC go bump. Let's readd it

File: main_test.go [hunk 4]
```diff
@@ -251,25 +260,31 @@ func Test_Main_SampleXML(t *testing.T) {
...
 	// 1. get the spans and metrics JSONs, they are separated by \n
 	// 2. remote white spaces
 	// 3. unmarshal each resource separately
 	// 4. assign each resource to the test report struct
-	content := string(jsonBytes)
 
-	jsons := strings.Split(strings.TrimSpace(content), "\n")
-	if len(jsons) != 2 {
-		t.Errorf("expected 2 JSONs, got %d - %s", len(jsons), jsons)
+	var jsons []string
```
Comment (mdelapenya) [inline main_test.go:269]: Suggestion: same here, let's readd the original changed

**Review by mdelapenya — CHANGES_REQUESTED** [review]
Once merged, I'll ping you in the PR bumping TC go for review.

In the meantime, let's address my comments about not using the retry.

File: main_test.go (hunk 2 above)
Comment (ryanrolds) [reply main_test.go:217]: This was a mistake. I'm juggling a few branches and goofed a merge.

**Comment by ryanrolds** [comment]
Should be fixed.

**Review by mdelapenya — APPROVED** [review]
LGTM, thanks!

**Comment by mdelapenya** [comment]
Merged, thanks! I'm submitting the bump of testcontainers, and some tests cleanups now

**Pull request merged**